src/main/java/org/truffleruby/language/LazyRubyNode.java

```diff
@@ -62,4 +62,20 @@ public RubyNode resolve() {
         });
     }
 
+    private RubyNode getOrCreateMasterResolution() {
+        synchronized (resolutionMaster) {
```

Comment:
This generates a FindBugs warning:
`M M JLM: Synchronization performed on java.util.concurrent.atomic.AtomicReference in org.truffleruby.language.LazyRubyNode.getOrCreateMasterResolution()  At LazyRubyNode.java:[line 66]`